### PR TITLE
[wenet] add randm mask with spans  batch mask

### PR DESCRIPTION
**Pull request by Mddct**
支持 mask with spans in batch way
- 可替代wav2vec etc等ssl 的mask
- 可替代 specaug 中的time mask 和freq mask
- batch 处理（可gpu）
- text inject的时候需要batch 的mask， 模拟音频的specaug

```python
t_sequence_mask((2, torch.tensor([20, 10]), mask_size=20, multiplicity=3, max_length=torch.tensor(4), fix_length=False)
tensor([[1., 1., 1., 1., 1., 0., 0., 0., 1., 0., 0., 0., 1., 1., 0., 1., 1., 1.,
         1., 1.],
        [1., 1., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
         1., 1.]])
t_sequence_mask((2, torch.tensor([20, 10]), mask_size=20, multiplicity=3, max_length=torch.tensor(4), fix_length=True)
tensor([[0., 0., 0., 1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 1., 0., 0., 0., 1.,
         1., 1.],
        [0., 0., 0., 0., 1., 0., 0., 0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
         1., 1.]])
```
- [x] time mask and freq mask

### CODE REVIEW

**Comment by xingchensong** [comment]
I think it is better to add a unit test for this function :) espnet和hugginface的unit test做的特比好，新函数/新文件 必带unittest，我们应该学习下

**Comment by Mddct** [comment]
hold this until have time
